llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -563,3 +600,21 @@ AIEHazardRecognizer::getSelectedAltOpcode(MachineInstr *MI) const {
     return It->second;
   return std::nullopt;
 }
+
+MemoryBankBits AIEHazardRecognizer::getMemoryBanks(MachineInstr *MI) const {
+  MemoryBankBits MemoryBankUsed = 0;
+
+  if (!(MI->mayLoad() || MI->mayStore()))
+    return MemoryBankUsed;
```

Comment:
Nit: return `0` directly, and `~0` below. We can then move `MemoryBankBits MemoryBankUsed = 0;` down, closer to where it is used.